cuda_bindings/docs/source/overview.md

```diff
@@ -291,22 +290,286 @@ nsys profile -s none -t cuda --stats=true <executable>
   - 1080ms
 ```
 
-CUDA Python is also compatible with [NVIDIA Nsight
+`cuda.bindings` is also compatible with [NVIDIA Nsight
 Compute](https://developer.nvidia.com/nsight-compute), which is an
 interactive kernel profiler for CUDA applications. It allows you to have
-detailed insights into kernel performance. This is useful when you’re trying to
+detailed insights into kernel performance. This is useful when you're trying to
 maximize performance ({numref}`Figure 1`).
 
-```{figure} _static/images/Nsigth-Compute-CLI-625x473.png
+```{figure} _static/images/Nsight-Compute-CLI-625x473.png
 :name: Figure 1
 
-Screenshot of Nsight Compute CLI output of CUDA Python example.
+Screenshot of Nsight Compute CLI output of `cuda.bindings` example.
+```
+
+## Preparing kernel arguments
+
+The `cuLaunchKernel` API bindings retain low-level CUDA argument preparation requirements:
+
+* Each kernel argument is a `void*` (i.e. pointer to the argument)
+* `kernelParams` is a `void**` (i.e. pointer to a list of kernel arguments)
+* `kernelParams` arguments are in contiguous memory
+
+These requirements can be met with two different approaches, using either NumPy or ctypes.
+
+### Using NumPy
+
+NumPy [Array objects](https://numpy.org/doc/stable/reference/arrays.html) can be used to fulfill each of these conditions directly.
+
+Let's use the following kernel definition as an example:
+```python
+kernel_string = """\
+struct testStruct {
+    int value;
+};
+
+extern "C" __global__
+void testkernel(int i, int *pi,
+                float f, float *pf,
+                struct testStruct s, struct testStruct *ps)
+{
+    *pi = i;
+    *pf = f;
+    ps->value = s.value;
+}
+"""
+```
+
+The first step is to create array objects with types corresponding to your kernel arguments. Primative Numpy types have the following corresponding kernel types:
+
+```{list-table} Correspondence between NumPy types and kernel types.
+:header-rows: 1
+
+* - NumPy type
+  - Corresponding kernel types
+  - itemsize (bytes)
+* - bool
+  - bool
+  - 1
+* - int8
+  - char, signed char
+  - 1
+* - int16
+  - short, signed short
+  - 2
+* - int32
+  - int, signed int
+  - 4
+* - int64
+  - long long, signed long long
+  - 8
+* - uint8
+  - unsigned char
+  - 1
+* - uint16
+  - unsigned short
+  - 2
+* - uint32
+  - unsigned int
+  - 4
+* - uint64
+  - unsigned long long
+  - 8
+* - float16
+  - half
+  - 2
+* - float32
+  - float
+  - 4
+* - float64
+  - double
+  - 8
+* - complex64
+  - float2, cuFloatComplex, complex&lt;float&gt;
+  - 8
+* - complex128
+  - double2, cuDoubleComplex, complex&lt;double&gt;
+  - 16
+```
+
+Furthermore, custom NumPy types can be used to support both platform-dependent types and user-defined structures as kernel arguments.
+
+This example uses the following types:
+* `int` is `np.uint32`
+* `float` is `np.float32`
+* `int*` and `float*` is `np.uint64`
+* `testStruct` is a custom user type `np.dtype([("value", np.int32)])`
```

Comment:
To be safe when the struct is more complex, we want to set `align=True`
```suggestion
* `testStruct` is a custom user type `np.dtype([("value", np.int32)], align=True)`
```
which would give us the same layout as a C compiler would produce.